includes/wikia/api/SendGridPostBackApiController.class.php

```diff
@@ -1,91 +1,115 @@
-
-		if (empty($events)) {
-			\Wikia\Logger\WikiaLogger::instance()->debug( __CLASS__ . ": No data to process");
-			wfProfileOut( __METHOD__ );
+		if ( !$this->wg->Request->wasPosted() ) {
+			// Log the token-validation problem.
+			WikiaLogger::instance()->debug( __CLASS__ . ': Request method must be POST' );
+			$this->response->setVal( 'response', 'Request method must be POST' );
 			return;
 		}
 
-		if (!is_array($events)) {
-			\Wikia\Logger\WikiaLogger::instance()->debug( __CLASS__ . ": Supplied data is not an array", [
-				'data_first_100_chars' => is_string($data) ? substr($data,0,100) : '[type: ' . gettype($data) . ']'
-			]);
-			wfProfileOut( __METHOD__ );
+		$events = $this->getEvents();
+		if ( empty( $events ) ) {
+			$this->response->setVal( 'response', 'Unable to find any events' );
 			return;
 		}
 
-		foreach ($events as $event) {
-			$email_addr = $this->safe_get($event, 'email', '');
-			$email_id 	= $this->safe_get($event, 'wikia-email-id');
-			$city_id 	= $this->safe_get($event, 'wikia-email-city-id');
-			$sender_db 	= $this->safe_get($event, 'wikia-db');
-			$token 		= $this->safe_get($event, 'wikia-token');
-			$event_type	= $this->safe_get($event, 'event');
-			$url 		= $this->safe_get($event, 'url', '');
-			$status 	= $this->safe_get($event, 'status', '');
-			$reason		= $this->safe_get($event, 'reason', '');
-
-			if ( $this->wg->Request->wasPosted() ) {
-				// log postback data
-				$insert_data = array(
-					"mail_id"      => $email_id,
-					"emailAddr"    => $email_addr,
-					"cityId"       => $city_id,
-					"eventType"    => $event_type,
-					"senderDbName" => $sender_db,
-					"url"          => $url,
-					"status"       => $status,
-					"reason"       => $reason
-				);
-
-				// Log to logstash/kibana
-				\Wikia\Logger\WikiaLogger::instance()->info( "SendgridPostback", $insert_data);
-
-				// On bounce, invalidate the users email to force them to re-verify it
-				if ( $event_type == 'bounce' ) {
-					if ( isset( $this->wg->SharedDB ) ) {
-						$dbr = wfGetDB( DB_SLAVE, array(), $this->wg->ExternalSharedDB );
-					} else {
-						$dbr = wfGetDB( DB_SLAVE );
-					}
-
-					$res = $dbr->select( array( '`user`' ), array( 'user_id' ), array( 'user_email' => $email_addr ), __METHOD__ );
-					while( $row = $dbr->fetchObject( $res ) ) {
-						$user = User::newFromId( $row->user_id );
-						if (!$user) {
-							continue;
-						}
-						$user->invalidateEmail();
-						$user->saveSettings();
-					}
-				}
-
-				$this->response->setVal( 'response', 'Postback processed.' );
+		foreach ( $events as $event ) {
+			$eventType = $this->safeGet( $event, 'event' );
+
+			// Log to logstash/kibana
+			WikiaLogger::instance()->info( 'SendgridPostback', $event );
+
+			// On bounce, invalidate the users email to force them to re-verify it
+			if ( $eventType == 'bounce' ) {
+				// SOC-1435 : For now don't invalidate email on bounce
+				# $this->handleBounce( $event );
```

Comment:
Only that its visually distinct from the comment above it, so it didn't look like it was part of the comment.  Otherwise I don't really feel strongly about it.